src/coreclr/gc/gc.cpp

```diff
@@ -6472,7 +6473,8 @@ class heap_select
         if (GCToOSInterface::CanGetCurrentProcessorNumber())
         {
             uint32_t proc_no = GCToOSInterface::GetCurrentProcessorNumber();
-            int adjusted_heap = proc_no_to_heap_no[proc_no];
+            // proc_no could likely exceed MAX_SUPPORTED_CPUS on x86
+            int adjusted_heap = proc_no_to_heap_no[proc_no % MAX_SUPPORTED_CPUS];
```

Comment:
I know this was code from before but do we need `adjusted_heap` to be signed? Will there ever be a case where adjusted_heap is negative?

Reply:
don't believe that could occur.